xls/dslx/stdlib/std.x

```diff
@@ -374,6 +374,95 @@ fn msb_test() {
     assert_eq(u1:1, msb(s2:0b11));
 }
 
+
+// Returns the R LSbs of x.
+// `x` must be at least R bits wide.
+pub fn lsbs<R: u32, N: u32>(x: uN[N]) -> uN[R] {
+  const_assert!(N >= R);
+  x[0 +: uN[R]]
+}
+
+#[test]
+fn lsbs_test() {
+    assert_eq(lsbs<u32:1>(u2:1), u1:1);
+    assert_eq(lsbs<u32:2>(u3:5), u2:1);
+    assert_eq(lsbs<u32:3>(u5:30), u3:6);
+}
+
+// Returns the R MSbs of x.
+// `x` must be at least R bits wide.
+pub fn msbs<R: u32, N: u32>(x: uN[N]) -> uN[R] {
+  const_assert!(N >= R);
+  x[(N - R) +: uN[R]]
+}
+
+#[test]
+fn msbs_test() {
+    assert_eq(msbs<u32:1>(u2:1), u1:0);
+    assert_eq(msbs<u32:2>(u3:5), u2:2);
+    assert_eq(msbs<u32:3>(u5:30), u3:7);
+}
+
+// Splits bits into (the remaining most significant bits, N least significant
+// bits).
+//
+// This function ensures that all bits of the argument are used.
+pub fn split_lsbs<N: u32, X: u32, Y: u32 = {X - N}, FROM_START: s32 = {N as s32}>
```

Comment:
nit (as this is not really something introduced by this change), but we should consider making `FROM_START` an const binding internal to the function since it's not really something meant for the user to override.